### PR TITLE
[96] Fix print to logging bugs

### PR DESCRIPTION
**Pull request by jpmcfarland**
Not only fix the bugs caused by the conversion, but make _all_ logging calls uniform in format.

~~It would be nice to (re)consolidate the two versions of `init_logger()`,  but that would require significant updates to `iBridgesCli.py` and is out of this scope of this issue.~~

Consolidating the `init_logger()` function is the subject of PR #102.

### CODE REVIEW

**Comment by jpmcfarland** [comment]
@wurDevTim @maartenschermer I know Christine is out for the next two weeks, but I want her to be aware.  I also would like reviews in the form of "none of the logging statements cause an error".  Everything looks fine to me, but I simply don't have the means to test everything.

**Review by wurDevTim — COMMENTED** [review]
Heet John, 
Looked over the changes, I had a few small notes which I put inline. 
Also pressed around in the GUI, that looked good.

File: gui/continousUpload.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def run(self):
                         folder_wfiles = self.tosync_dictionary.pop(filepath)
                         self.conn.upload_data(filepath, self.destColl, None, None, force=self.force)
                     else:
-                        logging.debug("Somethings going wrong. data folder in {filepath} not tracked")
+                        logging.debug("Something's going wrong. data folder in %s not tracked", filepath)
```
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply gui/continousUpload.py:62]: Stylistically, I use single quotes for all strings unless there is one or more within the string.  Escaping is possible, but not ideal, so I avoid it.  I don't even like `\`s to continue lines unless it is absolutely necessary.

File: gui/mainmenu.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 class QPlainTextEditLogger(logging.Handler):
     def __init__(self, widget):
-        super(QPlainTextEditLogger, self).__init__()
+        super().__init__()
```
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply gui/mainmenu.py:17]: Yes, I see.  Thanks for the catch.

File: irodsConnector/session.py [hunk 3]
```diff
@@ -170,19 +170,21 @@ def connect(self):
...
             else:
-                logging.error(f'{kw.RED}No iRODS session: "irods_env_file" not set!{kw.DEFAULT}')
+                logging.error(
+                    '%sNo iRODS session: "irods_env_file" not set!%s',
+                    kw.RED, kw.DEFAULT)
                 return
         options = {
             'irods_env_file': str(self.irods_env_file),
         }
-        logging.debug(f'{self.ienv=}')
```
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply irodsConnector/session.py:185]: Thanks!  I certainly did miss that one.

File: gui/mainmenu.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply gui/mainmenu.py:17]: Upon research, `QPlainTextEditLogger` is defined here.  So `super()` refers correctly to `logging.Handler`.

File: irodsConnector/session.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply irodsConnector/session.py:185]: No, I did remove that on purpose.  I want to use those statements only in select properties so I can see how often they are called.

File: gui/dataTransfer.py [hunk 4]
```diff
@@ -252,7 +252,7 @@ def run(self):
                                                    self.coll.path, newPath, scope="checksum")
                 self.updLabels.emit(len(onlyIrods), len(diff))
         except:
-            logging.exception("dataTransfer.py: Error in getDataState")
+            logging.exception('dataTransfer.py: Error in getDataState')
```
Comment (maartenschermer) [inline gui/dataTransfer.py:255]: Not sure if `logging.exception` is a deliberate choice here (over `logging.error`), but I think we'd better stick to `logging.error`. `logging.exception` generates additional lines of stacktrace that are not useful here.

File: utils/context.py
Comment (jtersteegwur) [inline utils/context.py:1]: Would it be possible to change the print on Line 264 to be logging as well?

https://github.com/chStaiger/iBridges-Gui/blob/17e7bffed8335dd88dd9a1de32e11a609b835622/utils/context.py#L264

File: gui/IrodsBrowser.py [hunk 5]
```diff
@@ -393,7 +393,7 @@ def fillInfo(self, index):
             self._fill_acls_tab(obj_path)
             self._fill_replicas_tab(obj_path)
         except Exception as e:
```
Comment (maartenschermer) [inline gui/IrodsBrowser.py:395]: For concistency's sake, `e` should be `error`, I think.

File: iBridges.py [hunk 6]
```diff
@@ -186,9 +186,9 @@ def login_function(self):
             self.setCursor(PyQt6.QtGui.QCursor(PyQt6.QtCore.Qt.CursorShape.ArrowCursor))
             return
         except Exception as unknown:
```
Comment (maartenschermer) [inline iBridges.py:188]: For concistency's sake, `e` should be `error`, I think.

File: iBridgesCli.py [hunk 7]
```diff
@@ -242,11 +242,10 @@ def connect_irods(cls, irods_env):
 
                 break
             except AssertionError as exception:
```
Comment (maartenschermer) [inline iBridgesCli.py:244]: For concistency's sake, `exception` should be `error`, I think.

File: utils/elab_plugin.py [hunk 8]
```diff
@@ -34,8 +34,9 @@ def setup(self, calling_class):
             try:
                 self.elab.updateMetadataUrl(group=config['group'], experiment=config['experiment'])
             except ValueError as exception:
-                logging.error("ELN groupID %s or experimentID %s not set or invalid. (%s)",
```
Comment (maartenschermer) [inline utils/elab_plugin.py:37]: For consistency's sake, I think `exception` should be `error` (occurs several more times below).

**Comment by jtersteegwur** [comment]
Along similar lines to @maartenschermer 's comments:
https://github.com/chStaiger/iBridges-Gui/blob/jpm/bugfix/96_print_to_logging/gui/irodsTreeView.py#L55

Could/should this be an error?

File: irodsConnector/dataOperations.py [hunk 9]
```diff
@@ -321,7 +321,7 @@ def upload_data(self, source: str, destination: irods.collection.iRODSCollection
         if not force:
             space = self.resc_man.resource_space(res_name)
             if size > (space - buff):
-                logging.info(
+                logging.error(
```
Comment (maartenschermer) [inline irodsConnector/dataOperations.py:324]: I think putting the actual word 'ERROR' in the error string is superfluous (several more instances below).

File: irodsConnector/meta.py [hunk 10]
```diff
@@ -28,9 +28,9 @@ def add(self, items: list, key: str, value: str, units: str = None):
             try:
                 item.metadata.add(key.upper(), value, units)
             except irods.exception.CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME:
-                logging.error(kw.RED+"INFO ADD META: Metadata already present"+kw.DEFAULT)
+                logging.error('%sADD META: Metadata already present%s', kw.RED, kw.DEFAULT)
```
Comment (maartenschermer) [inline irodsConnector/meta.py:31]: I think we should remove the colour codes (or at leat move somehow them to the logging handler).

File: irodsConnector/permission.py [hunk 11]
```diff
@@ -106,11 +106,10 @@ def set_permissions(self, perm: str, path: str, user: str = '',
                     self.data_man.collection_exists(path):
                 self.sess_man.irods_session.permissions.set(acl, recursive=recursive, admin=admin)
         except irods.exception.CAT_INVALID_USER as ciu:
```
Comment (maartenschermer) [inline irodsConnector/permission.py:108]: I think for consistency, `ciu` and `cia` should become `error`.

File: utils/context.py
Comment (jpmcfarland) [reply utils/context.py:1]: Yep!

File: iBridges.py (hunk 6 above)
Comment (jpmcfarland) [reply iBridges.py:188]: Consistency is nice.

File: utils/elab_plugin.py (hunk 8 above)
Comment (jpmcfarland) [reply utils/elab_plugin.py:37]: Got it.

File: irodsConnector/tickets.py [hunk 12]
```diff
@@ -52,7 +52,7 @@ def create_ticket(self, obj_path: str, expiry_string: str = '') -> tuple:
             try:
                 expiration_set = self._modify_ticket(ticket, expiry_string)
             except Exception as error:
-                logging.info('Could not set expiration date: %s', str(error))
+                logging.info('Could not set expiration date: %r', error)
```
[the remark this answers is not in the document]
Comment (jpmcfarland) [reply irodsConnector/tickets.py:55]: Absolutely.

File: irodsConnector/dataOperations.py (hunk 9 above)
Comment (jpmcfarland) [reply irodsConnector/dataOperations.py:324]: All updated.

File: irodsConnector/permission.py (hunk 11 above)
Comment (jpmcfarland) [reply irodsConnector/permission.py:108]: Makes sense as long as they are only used locally and not needed again.

File: irodsConnector/meta.py (hunk 10 above)
Comment (jpmcfarland) [reply irodsConnector/meta.py:31]: As far as I can tell, there is no per-type formatting, only one format per handler.  I think the best I can do is create a set of wrapper functions (or even a single function) to call the logger methods with the appropriate ASCII color codes surrounding the logging message.  Perhaps something like:

```python
def log(log_method, log_message):
    mdict = {
        logging.debug: '%s',
        logging.info: 'kw.YELLOW%skw.DEFAULT',
        logging.warning: 'kw.ORANGE%skw.DEFAULT',
        logging.error: 'kw.RED%skw.DEFAULT',
    }
    log_method(mdict[log_method], log_message)
```

Unfortunately, every single logging statement will have to be updated and the advantage of optional evaluation will be lost.

Comment (jpmcfarland) [reply irodsConnector/meta.py:31]: I think a simple string converting function is the best solution.  A little more work when writing the logging statement, but retains all the other advantages.

Comment (jpmcfarland) [reply irodsConnector/meta.py:31]: Found the `correct` solution: **add custom formatting targets to the record factory.**

Comment (jpmcfarland) [reply irodsConnector/meta.py]: See: https://github.com/chStaiger/iBridges-Gui/pull/97/commits/0098e934972b4916f2d29260b69714ca2c6d2a55

**Comment by jpmcfarland** [comment]
@jtersteegwur @maartenschermer @wurDevTim Please have another quick sanity check.

Approve or leave a message so I know you have seen it.  Thanks!

File: irodsConnector/meta.py (hunk 10 above)
Comment (maartenschermer) [reply irodsConnector/meta.py]: Looks good! Is it possible to put the `init_logger()` somewhere more central, maybe in utils, so `iBridgesCli.py` can use it too?

**Review by maartenschermer — APPROVED** [review]

**Pull request merged**